### DISCUSSION TITLE
What do you think about adding the AI Logo creation feature?

### DISCUSSION THREAD

**Discussion by ancy0** [AgainIoT/Open-Set-Go#41] [Feature Suggestion / Share Ideas]
**Many of the features we offer already exist, but I thought it would be good to add the AI Logo generation feature in the repository create step.**

**Of course, this AI Logo will be a temporary Logo for users.
Although it is a sample, this can make their project more memorable, and it can also be helpful when creating their own logo later on.**

**I want to know what you guys think about this.** 😀

우리가 제공하는 많은 기능들이 이미 존재하지만, 저장소 생성 단계에서 AI 로고 생성 기능을 추가하는 것도 나쁘지 않을 것이라고 생각합니다. 물론 이 AI 로고는 사용자들을 위한 임시 로고가 될 것입니다. 비록 샘플이지만, 이것은 그들의 프로젝트를 더 기억에 남을 수 있게 해줄 수 있고, 나중에 그들만의 로고를 만들 때도 도움이 될 수 있습니다. 여러분들은 이것에 대해 어떻게 생각하는지 알고 싶습니다.

**Comment by ymw0407** [comment]
I think it's a good idea! I think it would be good to find out if there is an API provided from outside regarding that part.
If other maintainers are looking for related APIs, it would be great to share them here!
좋은 아이디어인 것 같습니다! 해당 부분과 관련하여 외부에서 제공되는 API가 있는지 찾아보면 좋을 것 같습니다.
다른 Maintainer분들도 관련 API에 관해서 찾으신다면 여기에 공유해주시면 좋을 것 같습니다!

@AgainIoT/development-team

**Comment by hummingbbird** [comment]
I think that is a good idea👍 It is essential to create a logo that represents the project when you start it. After looking at @ymw0407 's comment, I found an API about AI Logo creation. please take a look at the api information attached below and leave your comments!
좋은 의견이라고 생각합니다👍 프로젝트를 시작할 때 프로젝트를 대표하는 로고를 생성하는 것은 필수적이기 때문입니다. 다른 maintainer의 comment를 보고 찾아본 결과 아래와 같은 api를 찾을 수 있었습니다. 다른 분들도 아래 첨부한 api 정보를 한 번 보고 의견을 남겨주세요!

https://www.logoai.com/logo-api

**Reply by ymw0407** [reply]
![image](https://github.com/AgainIoT/Open-Set-Go/assets/77202633/5664ae4d-8f63-4a20-b1c7-4d5f19046a1e)

Apparently that API is good too! But... Since the API itself is using a paid plan, I think it may be a little cautious to use it on OpenSource. First of all, keep that API on hold, and let me know if there is a good Logo generation API for other maintainers!
해당 API도 좋은 것 같습니다! 하지만... 해당 API 자체가 유료 플랜을 사용하고 있기 때문에 Open-Source에서 사용하기 위해서는 조금 조심스러울 수 있다는 생각이 듭니다. 우선 해당 API는 보류해두고, 다른 maintainer분들도 좋은 Logo 생성 API가 있다면 알려주세요!

**Comment by sinji2102** [comment]
Since I think it's a good opinion, I looked into AI-Logo-Generator that can be applied to the project. I am attaching a link related to this.
좋은 의견이라고 생각하기 때문에, 프로젝트에 응용할 수 있는 AI-Logo-Generator를 알아보았습니다. 관련하여 링크를 첨부합니다.

https://hotpot.ai/logo-generator

However, in order to apply these opinions, it is expected that there will be various problems such as copyright and intellectual property rights. I think we need to share the relevant opinions within the team.
그러나, 이러한 의견을 적용하기 위해서는 저작권과 지적재산권 등의 다양한 문제가 있을 것이라고 예상됩니다. 팀 내에서 관련 의견 공유가 필요할 것 같습니다.

**Reply by ymw0407** [reply]
As you said above, I agree that copyright and intellectual property issues should also be considered.
Unfortunately, however, it seems difficult to use it in the Open Source project because we are implementing a charge policy even in the case of the API. Thank you for your opinion! :smile:
위에서 말씀해주신 것처럼 저작권과 지적재산권 문제도 고려해야한다는 점에서 공감합니다.
하지만 아쉽게도, 해당 API와 같은 경우에도 유료화 정책을 시행중이기 때문에 Open Source 프로젝트에서 사용하기 힘든 것 같습니다. 의견 감사합니다! :smile:

**Comment by Kim-Jiyun** [comment]
Creating AI logos sounds like a good idea! If you make a logo and provide it, I think you will feel like starting a project in earnest. That's why I look up related api a lot and post appi that I think is okay. Please check it out and leave your comments!
AI logo를 생성해주는 것은 좋은 아이디어 같습니다! logo를 만들어서 제공해 준다면 본격적으로 프로젝트를 시작하는 느낌을 받을 수 있을 것 같습니다. 저도 그래서 관련 api를 많이 찾아보고 괜찮다고 생각하는 api를 올립니다. 확인하시고 의견 남겨주세요!

https://stablediffusionapi.com/docs/stable-diffusion-api/text2img

**Reply by ymw0407** [reply]
For the Stable Diffusion API, I know it's an algorithm that's unlocked with Open Source! If you use it, I think it would be good to use that API!!!
Stable Diffusion API와 같은 경우에는 Open Source로 풀려있는 알고리즘으로 알고 있습니다! 만약 사용한다면 해당 API를 사용하면 좋을 것 같습니다!!!

**Reply by ymw0407** [reply]
Oh, there's a problem. If you want to use that algorithm, you'll need a server computer that's very capable enough to utilize AI...In particular, Stable Diffusion API is unlikely to be available because we want to continue to operate the service and because the project is being used for non-profit purposes, even as an Open Source...
아.. 문제가 하나 있습니다. 해당 알고리즘을 사용하고자 한다면, AI를 활용할 수 있을 정도의 굉장히 성능이 좋은 서버 컴퓨터가 필요할 것 같습니다....특히, 저희가 해당 서비스를 지속적으로 운영하고자 한다는 점과 해당 프로젝트는 비영리적인 목적의, 심지어 Open-Source로 사용하고 있기 때문에 Stable Diffusion API를 사용하긴 힘들 것 같습니다...

**Comment by ymw0407** [comment]
After looking at the APIs surveyed by many maintainers, we decided that the AI Logo generator would be difficult for us to use. If a good API or algorithm to solve the above problems!If there is, it would be nice to reopen the Discussion...
많은 maintainer들이 조사한 API들을 본 결과, AI Logo 생성기는 저희가 활용하기에 힘들겠다는 판단을 내리게 되었습니다. 만약 위의 문제들을 해결할 수 있는 좋은 API나 알고리즘!이 있다면, 해당 Discussion을 다시 오픈해주면 좋을 것 같습니다...
